interfaces/cython/cantera/thermo.pyx

```diff
@@ -1786,6 +1785,143 @@ cdef class ThermoPhase(_SolutionBase):
             cdef CxxUnits units = self.thermo.standardConcentrationUnits()
             return Units.copy(units)
 
+    # methods for plasma
+    property Te:
+        """Get/Set electron Temperature [K]."""
+        def __get__(self):
+                return self.thermo.electronTemperature()
+
+        def __set__(self, value):
+            if not self._enable_plasma:
+                raise TypeError('This method is invalid for '
+                                f'thermo model: {self.thermo_model}.')
+            self.plasma.setElectronTemperature(value)
+
+    def set_discretized_electron_energy_distribution(self, levels, distribution):
+        """
+        Set electron energy distribution. When this method is used, electron
+        temperature is calculated from the distribution.
+
+        :param levels:
+            vector of electron energy levels [eV]
+        :param distribution:
+            vector of distribution
+        """
+        if not self._enable_plasma:
+            raise TypeError('This method is invalid for '
+                            f'thermo model: {self.thermo_model}.')
+        # check length
+        if (len(levels) != len(distribution)):
+            raise ValueError('Length of levels and distribution are different')
+
+        cdef np.ndarray[np.double_t, ndim=1] data_levels = \
+            np.ascontiguousarray(levels, dtype=np.double)
+        cdef np.ndarray[np.double_t, ndim=1] data_dist = \
+            np.ascontiguousarray(distribution, dtype=np.double)
+
+        self.plasma.setDiscretizedElectronEnergyDist(&data_levels[0],
+                                                     &data_dist[0],
+                                                     len(levels))
+
+    property n_electron_energy_levels:
+        """ Number of electron energy levels """
+        def __get__(self):
+            return self.plasma.nElectronEnergyLevels()
```

Comment:
This method needs the check for `self._enable_plasma` to avoid causing a segfault for non-plasma phases.